components/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -345,9 +347,21 @@ private void persistWreply(ResponseToken responseToken, HttpSession session) {
     private void sendToAuthenticationFramework(HttpServletRequest request, HttpServletResponse response,
                                                String sessionDataKey, SessionDTO sessionDTO) throws IOException {
 
-        String commonAuthURL = IdentityUtil.getServerURL(FrameworkConstants.COMMONAUTH, false, true);
+        String commonAuthURL;
+        try {
+            commonAuthURL = ServiceURLBuilder.create().addPath(FrameworkConstants.COMMONAUTH).build()
+                    .getAbsolutePublicURL();
+        } catch (URLBuilderException e) {
+            throw new IOException("Error occurred while building the commonauth URL during login.", e);
+        }
+
+        String selfPath;
+        try {
+            selfPath = ServiceURLBuilder.create().addPath(request.getRequestURI()).build().getRelativeInternalURL();
+        } catch (URLBuilderException e) {
+            throw new IOException("Error occurred while building the commonauth caller path URL during login.", e);
+        }
 
-        String selfPath = request.getRequestURI();
```

Comment:
Make sure to set the tenantDomain in the AuthenticationRequest object during login and logout flows

Reply:
addressed at https://github.com/wso2-extensions/identity-inbound-auth-sts/pull/92